langchain4j-core/src/main/java/dev/langchain4j/internal/ExceptionMapper.java

```diff
@@ -0,0 +1,55 @@
+package dev.langchain4j.internal;
+
+import dev.langchain4j.exception.HttpException;
+import dev.langchain4j.exception.RateLimitException;
+import dev.langchain4j.exception.UnrecoverableException;
+
+import java.nio.channels.UnresolvedAddressException;
+import java.util.concurrent.Callable;
+
+@FunctionalInterface
+public interface ExceptionMapper {
+
+    ExceptionMapper DEFAULT = new DefaultImpl();
+
+    static <T> T mappingException(Callable<T> action) {
+        return DEFAULT.withExceptionMapper(action);
+    }
+
+    default <T> T withExceptionMapper(Callable<T> action) {
+        try {
+            return action.call();
+        } catch (Exception e) {
+            throw mapException(e);
+        }
+    }
+
+    RuntimeException mapException(Exception e);
+
+    class DefaultImpl implements ExceptionMapper {
+
+        @Override
+        public RuntimeException mapException(Exception e) {
```

Comment:
I sketched the default exception mapping here. Not sure if we could do something much better than this (especially if we want to keep this generic), but we could introduce specific implementation if required by some models.